src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -64,48 +70,221 @@ public class NewAzureNetworkWatcherConnectionMonitorCommand : ConnectionMonitorB
         [ValidateNotNull]
         [ValidateRange(1, int.MaxValue)]
         public int DestinationPort { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Ip address of the connection monitor destination.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Ip address of the connection monitor destination.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Ip address of the connection monitor destination.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Ip address of the connection monitor destination.",
+            ParameterSetName = "SetByConnectionMonitorV1")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationAddress { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The list of test group.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The list of test group.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The list of test group.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The list of test group.",
+            ParameterSetName = "SetByConnectionMonitorV2")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNetworkWatcherConnectionMonitorTestGroupObject> TestGroup { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor output.",
+            ParameterSetName = "SetByResource")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor output.",
+            ParameterSetName = "SetByName")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor output.",
+            ParameterSetName = "SetByLocation")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The connection monitor output.",
+            ParameterSetName = "SetByConnectionMonitorV2")]
+        //TODO
+        //[ValidateNotNullOrEmpty]
+        public List<PSNetworkWatcherConnectionMonitorOutputObject> Output { get; set; }
```

Comment:
use array type, also, remove commented code

Reply:
I am validating this parameter and If I used array like [] , I can not use LINQ functions